presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/TableScanMatcher.java

```diff
@@ -39,9 +50,11 @@ public boolean matches(PlanNode node, Session session, Metadata metadata, Symbol
             TableScanNode tableScanNode = (TableScanNode) node;
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableScanNode.getTable());
             String actualTableName = tableMetadata.getTable().getTableName();
-            if (expectedTableName.equalsIgnoreCase(actualTableName)) {
-                return true;
+            boolean matches = expectedTableName.equalsIgnoreCase(actualTableName);
+            if (matches && predicate.isPresent() && tableScanNode.getOriginalConstraint() != null) {
```

Comment:
The "original constraint" is there just for printing as part of EXPLAIN, and intended for human consumption. What should be validated is the effective predicate in the chosen table layout.